### ISSUE TITLE
Nuxt doesn't merge `app.config` fields imported using namespace import

### ISSUE THREAD

**Issue by enkot** [nuxt/nuxt#24708]
### Environment

- Operating System: Linux
- Node Version:     v18.18.0
- Nuxt Version:     3.8.2
- CLI Version:      3.10.0
- Nitro Version:    2.8.1
- Package Manager:  npm@9.4.2
- Builder:          -
- User Config:      devtools, extends
- Runtime Modules:  -
- Build Modules:    -

### Reproduction

https://stackblitz.com/edit/github-bcxfrv

### Describe the bug

I need to merge app.config from different layers, and I use namespace imports to combine different parts of the config, e.g:
```ts
import * as pages from './config/pages'

export default defineAppConfig({
  pages,
})
```
Check the console:
Expected: 
```js
{ nuxt: { buildId: 'dev' }, pages: { foo: { nested: 1 }, bar: 2 } }
```
Actual: 
```js
{ nuxt: { buildId: 'dev' }, pages: { bar: 2 } }
```

### Additional context

It was working fine in 3.7.4.

Also works fine if change import from namespace to named:
```ts
import { foo } from './config/pages'

export default defineAppConfig({
  pages: {
    foo
  }
})
```

**Comment by AndrewBogdanovTSS** [comment]
@danielroe Hi, any updates regarding this issue? Thanks

**Comment by AndrewBogdanovTSS** [comment]
After deeper investigation it was discovered that it's not Nuxt issue, it's **defu**
I logged a separate bug here https://github.com/unjs/defu/issues/119

**Comment by pi0** [comment]
Thanks for reporting the issue. Regression is fixed in [defu@6.1.4](https://github.com/unjs/defu/releases/tag/v6.1.4) you can upgrade using `npx nuxi upgrade --force`

https://stackblitz.com/edit/github-bcxfrv-kd2hkz?file=package.json

**Comment by AndrewBogdanovTSS** [comment]
thanks @pi0 for swiftly addressing it!
